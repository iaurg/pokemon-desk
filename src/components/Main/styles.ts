import styled from 'styled-components'

export const Wrapper = styled.main`
  background-color: #fff;
  color: #000;
  padding: 3rem;
  text-align: left;
  max-width: 1200px;
  margin: 0 auto;
`

export const Title = styled.h1`
  font-size: 2.5rem;
`

export const Description = styled.h2`
  font-size: 2rem;
  font-weight: normal;
  margin-bottom: 1rem;
`

export const Content = styled.div`
  display: grid;
`

export const Listing = styled.div`
  margin-top: 2rem;
`

export const Hand = styled.div`
  margin-bottom: 1rem;
`

export const HandTitle = styled.h3`
  font-size: 2rem;
  font-weight: bold;
  margin-bottom: 1rem;
`
