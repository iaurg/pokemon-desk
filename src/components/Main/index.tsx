import HandList from 'components/HandList'
import PokemonList from 'components/PokemonList'
import Search from 'components/Search'
import { ListProvider } from 'context/useList'
import { Toaster } from 'react-hot-toast'
import * as S from './styles'

const Main = ({
  title = 'Pokedex',
  description = 'Search for pokemons and add into your hand'
}) => (
  <ListProvider>
    <Toaster />
    <S.Wrapper>
      <S.Content>
        <S.Hand>
          <S.HandTitle>Your hand ðŸ‘‹</S.HandTitle>
          <HandList />
        </S.Hand>
        <S.Listing>
          <S.Title>{title} ðŸ‘¾</S.Title>
          <S.Description>{description}</S.Description>
          <Search />
          <PokemonList />
        </S.Listing>
      </S.Content>
    </S.Wrapper>
  </ListProvider>
)

export default Main
