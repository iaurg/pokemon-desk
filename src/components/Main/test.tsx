import { render, screen } from '@testing-library/react'

import Main from '.'

describe('<Main />', () => {
  it('should render the heading', () => {
    const { container } = render(<Main />)

    expect(
      screen.getByRole('heading', { name: /pokemon/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the colors correctly', () => {
    const { container } = render(<Main />)

    expect(container.firstChild).toHaveStyle({ 'background-color': '#fff' })
  })

  it('should render pokedex and hand', () => {
    render(<Main />)

    expect(
      screen.getByRole('heading', { level: 3, name: /Your Hand/i })
    ).toBeInTheDocument()
  })
})
