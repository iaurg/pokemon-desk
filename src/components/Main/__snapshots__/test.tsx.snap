// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the heading 1`] = `
.c5 input {
  background-color: #f4f5f5;
  border: none;
  padding: 1.5rem;
  border-radius: 0.8rem;
  width: 100%;
}

.c0 {
  background-color: #fff;
  color: #000;
  padding: 3rem;
  text-align: left;
  max-width: 1200px;
  margin: 0 auto;
}

.c3 {
  font-size: 2.5rem;
}

.c4 {
  font-size: 2rem;
  font-weight: normal;
  margin-bottom: 1rem;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c6 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c7 {
  font-size: 2rem;
  font-weight: bold;
  margin-bottom: 1rem;
}

<main
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <h1
        class="c3"
      >
        Pokedex
      </h1>
      <h2
        class="c4"
      >
        Search for pokemons and add into your hand
      </h2>
      <main
        class="c5"
      >
        <input
          placeholder="Search for pokemons"
          type="text"
        />
      </main>
    </div>
    <div
      class="c6"
    >
      <h3
        class="c7"
      >
        Your hand
      </h3>
    </div>
  </div>
</main>
`
